include/af/cuda.h

```diff
@@ -117,5 +122,93 @@ static inline void setNativeId(int nativeId)
 }
 #endif
 
+#if AF_API_VERSION >= 34
+/**
+   Interop function that converts a thrust::device_vector to an af::array
+
+   \param[in] tvec thrust::device_vector to be converted
+   \return af::array with converted contents of the thrust vector
+
+   \ingroup cuda_mat
+ */
+template<typename T>
+static af::array array(thrust::device_vector<T> &tvec)
+{
+    af::array output;
+    if(tvec.empty()) return output;
+    T* ptr = thrust::raw_pointer_cast(tvec.data());
+    af::array tmp = af::array(tvec.size(), ptr, afDevice);
+    output = tmp.copy();
+    tmp.lock();
+    return output;
+}
+#endif
+
+#if AF_API_VERSION >= 34
+/**
+   Interop function that converts a thrust::host_vector to an af::array
+
+   \param[in] tvec thrust::host_vector to  be converted
+   \return af::array with converted contents of the thrust vector
+
+   \ingroup cuda_mat
+ */
+template<typename T>
+static af::array array(thrust::host_vector<T> &tvec)
+{
+    af::array output;
+    if(tvec.empty()) return output;
+    //af_dtype type = (af_dtype)dtype_traits<T>::af_type;
+    T* ptr = thrust::raw_pointer_cast(tvec.data());
+    output = af::array(tvec.size(), ptr);
+    return output;
+}
+#endif
+
+#if AF_API_VERSION >= 34
+/**
+   Interop function that converts an af::array to a thrust::host_vector
+
+   \param[in] arr af::array to be converted to thrust::host_vector
+   \return thrust::host_vector with converted contents of af::array
+
+   \ingroup cuda_mat
+ */
+template<typename T>
+static thrust::host_vector<T> toHostVector(af::array &arr)
+{
+    af_dtype vec_type = (af_dtype)af::dtype_traits<T>::af_type;
+    if(arr.type() != vec_type) { throw af::exception("Thrust vector data type and array data type are mismatching"); }
+    //todo check continuity
+    thrust::host_vector<T> hvec(arr.elements());
+    T* h_ptr = thrust::raw_pointer_cast(hvec.data());
+    arr.host(h_ptr);
```

Comment:
`.host()` does the proper transformations to make it linear. You don't need to check for continuity.